dataset/src/main/scala/frameless/TypedEncoder.scala

```diff
@@ -27,23 +29,26 @@ abstract class TypedEncoder[T](implicit val classTag: ClassTag[T]) extends Seria
 //   - import TypedEncoder.usingInjection
 //   - import TypedEncoder.usingDerivation
 // """)
-object TypedEncoder {
+object TypedEncoder extends TypedEncoder0 {
```

Comment:
Yes, it is for priority. I had added prioritization while trying to figure out `Map` and it seemed useful to avoid ambiguity here (when injection would essentially always be preferred over derivation if it were available).

I can revert this change if you like.